local/compose.go

```diff
@@ -65,6 +70,100 @@ func (s *composeService) Build(ctx context.Context, project *types.Project) erro
 	return s.build(ctx, project, opts)
 }
 
+func (s *composeService) Push(ctx context.Context, project *types.Project) error {
+	configFile, err := config.Load(config.Dir())
+	if err != nil {
+		return err
+	}
+	eg, ctx := errgroup.WithContext(ctx)
+
+	info, err := s.apiClient.Info(ctx)
+	if err != nil {
+		return err
+	}
+	if info.IndexServerAddress == "" {
+		info.IndexServerAddress = registry.IndexServer
+	}
+
+	for _, service := range project.Services {
+		if service.Build == nil {
+			continue
+		}
+
+		eg.Go(func() error {
+			w := progress.ContextWriter(ctx)
+
+			ref, err := reference.ParseNormalizedNamed(service.Image)
+			if err != nil {
+				return err
+			}
+
+			repoInfo, err := registry.ParseRepositoryInfo(ref)
+			if err != nil {
+				return err
+			}
+
+			key := repoInfo.Index.Name
+			if repoInfo.Index.Official {
+				key = info.IndexServerAddress
+			}
+			authConfig, err := configFile.GetAuthConfig(key)
+			if err != nil {
+				return err
+			}
+
+			buf, err := json.Marshal(authConfig)
+			if err != nil {
+				return err
+			}
+
+			stream, err := s.apiClient.ImagePush(ctx, service.Image, moby.ImagePushOptions{
+				RegistryAuth: base64.URLEncoding.EncodeToString(buf),
+			})
+			if err != nil {
+				return err
+			}
+			dec := json.NewDecoder(stream)
+			for {
+				var jm jsonmessage.JSONMessage
+				if err := dec.Decode(&jm); err != nil {
+					if err == io.EOF {
+						break
```

Comment:
Wouldn't this return the error? Don't we want to `return nil` here?

Reply:
nope, this means end of the json events stream

Reply:
Yeah but the end of the stream is not an error right?